pkg/cli/jobserver/commands/serve.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	rbacv1 "github.com/kralicky/jobserver/pkg/apis/rbac/v1"
+	"github.com/kralicky/jobserver/pkg/auth"
+	"github.com/kralicky/jobserver/pkg/cgroups"
+	"github.com/kralicky/jobserver/pkg/jobs"
+	"github.com/kralicky/jobserver/pkg/rbac"
+	"github.com/kralicky/jobserver/pkg/server"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+	"sigs.k8s.io/yaml"
+)
+
+// ServeCmd represents the serve command
+func BuildServeCmd() *cobra.Command {
+	var rbacConfigFile string
+	var serverConfig server.Options
+	cmd := &cobra.Command{
+		Use:   "serve",
+		Short: "A brief description of your command",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if os.Geteuid() != 0 {
+				return fmt.Errorf("server must be run as root (try 'sudo %s serve')", os.Args[0])
+			}
+			if rbacConfigFile != "" {
+				config, err := loadRbacConfig(rbacConfigFile)
+				if err != nil {
+					return fmt.Errorf("failed to parse RBAC configuration: %w", err)
+				}
+				serverConfig.AuthMiddlewares = []auth.Middleware{
+					auth.NewMiddleware(auth.NewMTLSAuthenticator()),
```

Comment:
Isn't the mTLS authentication middleware a hard requirement to use `AuthenticatedUserFromContext` and the likes, under penalty of panic? Why is `rbacConfigFile` considered optional?

Reply:
Good point, I will make this required.